crates/flox-rust-sdk/src/models/publish.rs

```diff
@@ -358,4 +513,72 @@ mod tests {
 
         println!("{}", serde_json::to_string_pretty(&value).unwrap());
     }
+
+    #[cfg(feature = "impure-unit-tests")] // disabled for offline builds, TODO fix tests to work with local repos
+    #[tokio::test]
+    async fn git_file_error_if_dirty() {
+        use std::fs;
+        use std::path::Path;
+        use std::str::FromStr;
+        env_logger::init();
+
+        let (flox, _temp_dir_handle) = flox_instance();
+        let repo_dir = _temp_dir_handle.path().join("repo");
+
+        fs::create_dir(&repo_dir).unwrap();
+        let repo = Git::init(&repo_dir, false).await.unwrap();
+
+        fs::write(repo_dir.join("flake.nix"), "{ outputs = _: {}; }").unwrap();
+        repo.add(&[Path::new(".")]).await.unwrap();
+
+        let flake_ref =
+            GitRef::from_str(&format!("git+file://{}", repo_dir.to_string_lossy())).unwrap();
+
+        assert!(matches!(
+            PublishFlakeRef::from_git_file_flake_ref(
+                flake_ref.clone(),
+                false,
+                false,
+                &flox.nix(Default::default())
+            )
+            .await,
+            Err(ConvertFlakeRefError::LocalFlakeDirty)
+        ));
+    }
+
+    #[cfg(feature = "impure-unit-tests")] // disabled for offline builds, TODO fix tests to work with local repos
+    #[tokio::test]
+    async fn git_file_error_if_dirty2() {
+        use std::fs;
+        use std::path::Path;
+        use std::str::FromStr;
+        env_logger::init();
+
+        let (flox, _temp_dir_handle) = flox_instance();
+        let repo_dir = _temp_dir_handle.path().join("repo");
+
+        fs::create_dir(&repo_dir).unwrap();
+        let repo = Git::init(&repo_dir, false).await.unwrap();
+
+        fs::write(repo_dir.join("flake.nix"), "{ outputs = _: {}; }").unwrap();
+        repo.add(&[Path::new(".")]).await.unwrap();
+        repo.commit("Commit flake").await.unwrap();
+        repo.add_remote("upstream", &repo_dir.to_string_lossy())
+            .await
+            .unwrap();
+        repo.fetch().await.unwrap();
+        repo.set_origin("master", "upstream").await.unwrap();
```

Comment:
I usually don't have an upstream branch configured for my git branches, so I would add a test case that makes sure everything still works in that case

Reply:
added a test case for no upstream set